fastlane_core/lib/fastlane_core/android_package_name_guesser.rb

```diff
@@ -0,0 +1,65 @@
+module FastlaneCore
+  class AndroidPackageNameGuesser
+    class << self
+      def android_package_name_arg?(gem_name, arg)
+        return arg == "--package_name" ||
+               arg == "--app_package_name" ||
+               (arg == '-p' && gem_name == 'supply') ||
+               (arg == '-a' && gem_name == 'screengrab')
+      end
+
+      def guess_package_name_from_args(gem_name, args)
+        # args example: ["-a", "com.krausefx.app"]
+        package_name = nil
+        args.each_with_index do |current, index|
+          if android_package_name_arg?(gem_name, current)
+            package_name = args[index + 1] if args.count > index
```

Comment:
Might be worth briefly mentioning here why the `+1`, it's probably because of the command prefix?

Reply:
This is because of the way the arguments are passed in - in an array where the argument name is at [index] and then the value for the argument is at [index+1].

I'll put a comment on it!

Reply:
Sounds great, thanks, also at https://github.com/fastlane/fastlane/pull/10477#discussion_r142168913 I guess :)